gen3-integration-tests/tests/test_graph_submit_and_query.py

```diff
@@ -60,42 +95,436 @@ def test_submit_query_and_delete_records(self):
+            )
+            raise
+        self.sd_tools.delete_record(unique_id=record.unique_id)
+
+    @pytest.mark.graph_query
+    def test_with_path_to_first_to_last_node(self):
+        """
+        Scenario: Test with_path_to - first to last node
+        Steps:
+            1. Submit graph data using sheepdog. Verify records are present.
+            2. Query path from first node to last node.
+            3. Verify first node name is present response.
+        """
+        logger.info("Submitting test records")
+        self.sd_tools.submit_all_test_records()
+        # Query path from first node to last node.
+        first_node = self.sd_tools.test_records[self.sd_tools.submission_order[0]]
+        last_node = self.sd_tools.test_records[self.sd_tools.submission_order[-1]]
+        response = self.sd_tools.query_record_with_path_to(first_node, last_node)
+        # Verify first node name is present response.
+        assert (
+            first_node.node_name in response["data"].keys()
+        ), "{} not found in response {}".format(first_node.node_name, response)
+
+    @pytest.mark.graph_query
+    def test_with_path_to_last_to_first_node(self):
+        """
+        Scenario: Test with_path_to - last to first node
+        Steps:
+            1. Submit graph data using sheepdog. Verify records are present.
+            2. Query path from last node to first node.
+            3. Verify last node name is present response.
+        """
+        logger.info("Submitting test records")
+        self.sd_tools.submit_all_test_records()
+        # Query path from last node to first node.
+        first_node = self.sd_tools.test_records[self.sd_tools.submission_order[0]]
+        last_node = self.sd_tools.test_records[self.sd_tools.submission_order[-1]]
+        response = self.sd_tools.query_record_with_path_to(last_node, first_node)
+        # Verify last node name is present response.
+        assert (
+            last_node.node_name in response["data"].keys()
+        ), "{} not found in response {}".format(last_node.node_name, response)
+
+    @pytest.mark.skipif(
+        skip_consent_code_test(sd_tools),
+        reason="Consent Codes not available in dictionary",
+    )
+    def test_submit_data_record_with_consent_codes(self):
+        """
+        Scenario: Update file with invalid property
+        Steps:
+            1. Retrieve records from Indexd
+            2. Delete all records retrieved.
+            3. Submit metatdata for file node, including consent codes
+            4. Verify indexd record was created with the correct consent codes
+        """
+        indexd_auth = Gen3Auth(refresh_token=pytest.api_keys["indexing_account"])
+        gen3_indexd = Gen3Index(auth_provider=indexd_auth)
+        indexd = Indexd()
+        records = gen3_indexd.get_all_records()
+
+        # Delete indexd records
+        for record in records:
+            gen3_indexd.delete_record(guid=record["did"])
```

Comment:
why are we deleting all indexd records? this is inconvenient for anyone running tests on a dev/qa environment

Reply:
Not sure how big of an impact it would be, but as part of the gen3-qa test it was required to do it
https://github.com/uc-cdis/gen3-qa/blob/0a92cb47b62baa36b967f1dc34513de80aa72e95/suites/sheepdogAndPeregrine/submitAndQueryNodesTest.js#L209

Reply:
reading through the test, i don't see why we would need to do this. let's remove it and see if something breaks